omf/fileio/geoh5.py

```diff
@@ -181,52 +232,73 @@ class ElementConversion(BaseConversion):
     :obj:`geoh5py.objects.Points`
     """
 
-    _attribute_map = {
-        "description": "description",
+    _attribute_map: dict[str, Any] = {
         "name": "name",
         "uid": "uid",
         "geometry": None,
     }
 
     def __init__(self, obj: UidModel | Entity, geoh5: str | Path | Workspace):
         super().__init__(obj, geoh5)
-        self.geoh5_type = _CLASS_MAP[type(self.element.geometry)]
+
+        if isinstance(obj, UidModel):
+            self.geoh5_type = _CLASS_MAP[type(self.element.geometry)]
 
     def from_omf(self, **kwargs) -> Entity | None:
         """Convert omf element to geoh5 entity."""
         with fetch_h5_handle(self.geoh5) as workspace:
             try:
-                kwargs = self.collect_attributes(**kwargs)
+                kwargs = self.collect_omf_attributes(**kwargs)
             except OMFtoGeoh5NotImplemented as error:
                 warnings.warn(error.args[0])
                 return None
 
             self._entity = workspace.create_entity(
                 self.geoh5_type, **{"entity": kwargs}
             )
-            self.process_dependents(workspace)
+            self.process_dependents(self.element, workspace)
 
         return self._entity
 
-    def process_dependents(self, workspace):
-        if getattr(self.element, "data", None):
-            for child in self.element.data:
+    def from_geoh5(self, **kwargs) -> UidModel:
+        """Convert a geoh5 entity to omf element."""
+        with fetch_h5_handle(self.geoh5) as workspace:
+            kwargs = self.collect_h5_attributes(**kwargs)
+            uid = kwargs.get("uid")
```

Comment:
It's like I _always_ say Dom:

"When the pigs try and get at you..."

"`pop` it like it's Haaawt, `pop` it like it's` haaawt."

Reply:
ahaha lol. Somehow I pictured John Oliver making that joke